src/library/autodj/autodjfeature.cpp

```diff
@@ -271,21 +276,32 @@ void AutoDJFeature::slotCrateAutoDjChanged(int crateId, bool added) {
     }
 
     // Let subscribers know whether it's possible to add a random track.
-    emit(enableAddRandom(m_crateList.length() > 0));
+    bool addLibTracks = m_pConfig->getValueString(ConfigKey("[Auto DJ]", "IncludeLibTracks")).toInt();
+    emit(enableAddRandom(m_crateList.length() > 0 || addLibTracks));
 #endif // __AUTODJCRATES__
 }
 
 void AutoDJFeature::slotAddRandomTrack(bool) {
 #ifdef __AUTODJCRATES__
     // Get access to the auto-DJ playlist.
+    // **delete** We expect the caller to check if the track returned is actually available
+    //            Incase of those from library ,this method can determine whetther a returned
+    //            track is existing or not though such a check wont be exhaustive .(suggestions ?)
```

Comment:
Yes, the "is availabe" check should be done inside the called functions getRandomTrackId() should only return valid tracks